src/main/java/org/testng/xml/XmlTest.java

```diff
@@ -582,7 +582,7 @@ public String toXml(String indent) {
   @Override
   public String toString() {
 //    return toXml("");
-    StringBuffer result = new StringBuffer("[Test: \"" + m_name + "\"")
+    StringBuilder result = new StringBuilder("[Test: \"" + m_name + "\"")
```

Comment:
Do you have data showing that String#format is slow? Do you have performance-critical code paths in TestNG that you are aware of? Do you have a suite of performance tests or JMH benchmarks to identify performance regressions? Do you have performance requirements or a throughput SLA that you can articulate?